lms/djangoapps/courseware/courses.py

```diff
@@ -442,11 +443,12 @@ def get_course_assignment_due_dates(course, user, request, num_return=None, incl
 
                 block_url = None
                 now = datetime.now().replace(tzinfo=pytz.UTC)
-                assignment_released = item.start < now
+                assignment_released = item.start < now if item.start else None
                 if assignment_released:
                     block_url = reverse('jump_to', args=[course.id, block_key])
                     block_url = request.build_absolute_uri(block_url) if request else None
-                date_block.set_title(item.display_name, block_url)
+                assignment_title = item.display_name if item.display_name else _('Assignment')
+                date_block.set_title(assignment_title, block_url)
```

Comment:
This should be `date_block.set_title(assignment_title, link=block_url)` here right?